### PR TITLE
feat: Refactor serialization schema for performance.

### PR DESCRIPTION
**Pull request by doitian**
This PR contains 3 commits. The first two update the schema:

- Use struct instead of table to serialize on-chain data to make the
  serialized data more compact.
- Reorder the fields to make the structure alignment friendly.

The last one regenerates code from the schema and updates all the hashes.

### CODE REVIEW

**Comment by doitian** [comment]
The ckb-system-scripts crate is not published yet, blocked by the broken test.

**Comment by yangby-cryptape** [comment]
> The ckb-system-scripts crate is not published yet, blocked by the broken test.

The [CKB-System-Scripts PR-41: chore(deps): update ckb riscv docker and ckb crates](https://github.com/nervosnetwork/ckb-system-scripts/pull/41) is waiting for CI.

File: util/types/schemas/ckb.mol [hunk 1]
```diff
@@ -155,38 +155,38 @@ struct TransactionKey {
 }
 
 table TransactionInfo {
-    key:            TransactionKey,
```
Comment (u2) [inline util/types/schemas/ckb.mol]: Make the `TransactionInfo` as struct?

File: util/types/schemas/ckb.mol [hunk 2]
```diff
@@ -155,38 +155,38 @@ struct TransactionKey {
 }
 
 table TransactionInfo {
```
Comment (u2) [inline util/types/schemas/ckb.mol]: Make the `TransactionInfo` as struct?

Comment (doitian) [reply util/types/schemas/ckb.mol]: It's not an on-chain data structure, which is beyond the scope of this PR.

**Comment by u2** [comment]
bors r=quake,u2

**Comment by jjyr** [comment]
Please rebase

**Comment by u2** [comment]
bors r=quake,u2,yangby-cryptape,keroro520

**Comment by doitian** [comment]
bors retry

**Pull request merged**